test/logflare/backends/adaptor/slack_adaptor_test.exs

```diff
@@ -0,0 +1,265 @@
+defmodule Logflare.Backends.SlackAdaptorTest do
+  use Logflare.DataCase, async: false
+
+  alias Logflare.Backends.Adaptor.SlackAdaptor
+
+  doctest Logflare.Backends.Adaptor.SlackAdaptor, import: true
+
+  describe "send_message/1" do
+    test "can work with alert query" do
+      alert = insert(:alert)
+
+      Logflare.Backends.Adaptor.SlackAdaptor.Client
+      |> expect(:send, fn _url, body ->
+        assert body.blocks == [
+                 %{
+                   type: "section",
+                   accessory: %{
+                     type: "button",
+                     text: %{type: "plain_text", text: "Manage"},
+                     url: "http://localhost:4000/alerts/#{alert.id}",
+                     style: "primary"
+                   },
+                   text: %{
+                     type: "mrkdwn",
+                     text: "🔊 *#{alert.name}* | http://localhost:4000/alerts/#{alert.id}"
+                   }
+                 }
+               ]
+
+        {:ok, %Tesla.Env{}}
+      end)
+
+      assert {:ok, _} = SlackAdaptor.send_message(alert, [])
+
+      Logflare.Backends.Adaptor.SlackAdaptor.Client
+      |> expect(:send, fn _url, body ->
+        assert body.blocks == [
```

Comment:
lets send the body to the test proc to perform the assertions in the test case, otherwise errors in the separate proc won't cause the test to fail.

Reply:
we use this pattern quite extensively in other tests

Reply:
Got it, done in last pushed commit :pray: Thank you :rocket: